crates/pipeline_manager/src/db/mod.rs

```diff
@@ -1385,4 +1510,80 @@ impl ProjectDB {
         txn.commit().await?;
         Ok(())
     }
+
+    /// Retrieves the service identifier using its name, and uses this
+    /// identifier in the creation of a new service probe.
+    pub async fn new_service_probe_with_service_name(
+        &self,
+        tenant_id: TenantId,
+        service_name: &str,
+        id: Uuid,
+        request: ServiceProbeRequest,
+        created_at: &DateTime<Utc>,
+    ) -> Result<ServiceProbeId, DBError> {
+        let mut manager = self.pool.get().await?;
+        let txn = manager.transaction().await?;
+        let service = self
+            .get_service_by_name(tenant_id, service_name, Some(&txn))
+            .await?;
+        let service_probe_id = self
+            .new_service_probe(
+                tenant_id,
+                service.service_id,
+                id,
+                request,
+                created_at,
+                Some(&txn),
+            )
+            .await?;
+        txn.commit().await?;
+        Ok(service_probe_id)
+    }
+
+    /// Retrieves the service identifier using its name, and uses this
+    /// identifier to fetch the specific service probe.
+    pub(crate) async fn get_service_probe_with_service_name(
+        &self,
+        tenant_id: TenantId,
+        service_name: &str,
+        service_probe_id: ServiceProbeId,
+    ) -> Result<ServiceProbeDescr, DBError> {
+        let mut manager = self.pool.get().await?;
+        let txn = manager.transaction().await?;
+        let service = self
+            .get_service_by_name(tenant_id, service_name, Some(&txn))
+            .await?;
+        let descr = self
+            .get_service_probe(tenant_id, service.service_id, service_probe_id, Some(&txn))
+            .await?;
+        txn.commit().await?;
+        Ok(descr)
+    }
+
+    /// Retrieves the service identifier using its name, and uses this
```

Comment:
Similar comment for other methods here as well. Describe the contract, not the implementation unless necessary. So it's sufficient to say "Retrieves the list of service probes for the service named `service_name`".

Reply:
Definitely, will update the comments, thanks :)